network/lite/src/protocol.rs

```diff
@@ -0,0 +1,60 @@
+use std::time::SystemTime;
+
+pub const SYN_ACK_WAIT_MS: u32 = 500;
+pub const SYN_ACK_RETRIES: u32 = 5;
+pub const CONNECTION_EXPIRY_TIME_S: u32 = 5;
+pub const PAYLOAD_MAX_SIZE: usize = 512;
+
+pub enum WriteResult {
+  Success,
+  Failure_NotConnected,
+}
+
+#[derive(Clone, PartialEq)]
+pub enum ClientNegotiationStatus {
+  ClientSyn,
+  ServerSynAck,
+}
+
+#[derive(Clone, PartialEq)]
+pub enum ServerNegotiationStatus {
+  ClientSyn { last_message: SystemTime },
```

Comment:
This is probably better named as "ServerSynAck", to go with the client naming.

That is, it reflects the last message the owner sent.